fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanApplicationTerms.java

```diff
@@ -1871,4 +1871,16 @@ public LocalDate getRepaymentStartDate() {
                 : getSubmittedOnDate();
     }
 
+    public boolean isLastInstallment(final Integer installmentNumber) {
+        Integer numberOfRepayments = getNumberOfRepayments();
+        if (isDownPaymentEnabled()) {
```

Comment:
My gut feeling this is wrong...but i am not fully sure.
Can you please explain a little bit what was the idea here?

Reply:
Using the installment number for the last installment is different when you are using Down Payment (It creates a new installment) so we need to consider that when we need to identify the last installment

Reply:
There could be multiple disbursement and multiple down payment, so adding +1 is not right...

Reply:
Also while the the installment number is increased when a down payment period added, the period number is not. So you might wanna use that to decide on whether it is the last or not...